src/ring_theory/valuation/valuation_subring.lean

```diff
@@ -194,35 +194,34 @@ def ideal_of_le (R S : valuation_subring K) (h : R ≤ S) : ideal R :=
 instance prime_ideal_of_le (R S : valuation_subring K) (h : R ≤ S) :
   (ideal_of_le R S h).is_prime := (local_ring.maximal_ideal S).comap_is_prime _
 
+noncomputable
 def of_prime (A : valuation_subring K) (P : ideal A) [P.is_prime] :
   valuation_subring K :=
-of_le A (localization.subalgebra K P.prime_compl $
+of_le A (localization.subalgebra.of_field K P.prime_compl $
   le_non_zero_divisors_of_no_zero_divisors $ not_not_intro P.zero_mem).to_subring $
-  λ a ha, ⟨algebra_map A _ ⟨a,ha⟩, is_localization.lift_eq _ _⟩
+  λ a ha, subalgebra.algebra_map_mem _ (⟨a,ha⟩ : A)
 
+noncomputable
 instance of_prime_algebra (A : valuation_subring K) (P : ideal A) [P.is_prime] :
-  algebra A (A.of_prime P) :=
-show algebra A (localization.subalgebra _ _ _), by apply_instance
+  algebra A (A.of_prime P) := subalgebra.algebra _
```

Comment:
After `subalgebra` is replaced by `subalgebra.of_field`, `show` or `change` produces an obligation to proof that P.prime_compl is contained in non_zero_divisors (not sure why!), making the original proofs fail, so I replaced them with explicit terms.